### DISCUSSION TITLE
Help updating IHttpTransferManager from v2

### DISCUSSION THREAD

**Discussion by elmorg** [shinyorg/shiny#1347] [Q&A]
Hi,

I'm updating to the new version of Shiny and I used to have the line:
```
public IHttpTransferManager httpManager = ShinyHost.Resolve<IHttpTransferManager>();
```

I've now replaced this with the following line (as per the docs):
```
readonly IHttpTransferManager httpManager;
```

However, I'm getting the error `Object reference not set to an instance of an object.` when I run my app on Android. I've followed all the other setup instructions in the docs and have scoured the sample code to try and understand where I'm going wrong.

I think I understand the error, but it isn't clear to me how I should instantiate my `httpManager`. 

Any help or pointers would be greatly appreciated.

**Comment by aritchie** [comment]
ShinyHost doesn't exist in v3.  You also can't just replace a variable if you don't set it.  If you're using MAUI, it was meant to be injected into your viewmodel.  

If you don't have MAUI, you can use Shiny.Hosting.Host.GetService to get the service you need.
